python/pyspark/ml/util.py

```diff
@@ -523,11 +534,29 @@ def getAndSetParams(instance, metadata):
         """
         Extract Params from metadata, and set them in the instance.
         """
+        # User-supplied param values
         for paramName in metadata['paramMap']:
             param = instance.getParam(paramName)
             paramValue = metadata['paramMap'][paramName]
             instance.set(param, paramValue)
 
+        # Default param values
+        majorAndMinorVersions = majorMinorVersion(metadata['sparkVersion'])
+        assert majorAndMinorVersions is not None, "Error loading metadata: Expected " + \
+            "Spark version string but found {}".format(metadata['sparkVersion'])
+
+        major = majorAndMinorVersions[0]
+        minor = majorAndMinorVersions[1]
+        # For metadata file prior to Spark 2.4, there is no default section.
+        if major > 2 or (major == 2 and minor >= 4):
+            assert 'defaultParamMap' in metadata, "Error loading metadata: Expected " + \
+                "`defaultParamMap` section not found"
+
+            for paramName in metadata['defaultParamMap']:
+                param = instance.getParam(paramName)
+                paramValue = metadata['defaultParamMap'][paramName]
+                instance._setDefault(**{param.name: paramValue})
```

Comment:
remove line `param = instance.getParam(paramName)` and change this line to
`instance._setDefault(**{paramName: paramValue})`

Reply:
Oh, right. My bad.